java/core/src/java/org/apache/orc/OrcUtils.java

```diff
@@ -53,10 +57,14 @@
     }
     if (selectedColumns != null &&
         schema.getCategory() == TypeDescription.Category.STRUCT) {
+
       List<String> fieldNames = schema.getFieldNames();
       List<TypeDescription> fields = schema.getChildren();
+
+      boolean isAcid = SchemaEvolution.checkAcidSchema(schema);
+
       for (String column: selectedColumns.split((","))) {
-        TypeDescription col = findColumn(column, fieldNames, fields);
+        TypeDescription col = findColumn((isAcid ? "row." + column : column), fieldNames, fields);
```

Comment:
This code should ideally be in the Hive codebase and ORC should interpret the selectedColumns there.

Reply:
ACID schema generation code is under ORC SchemaEvolution class

public static TypeDescription createEventSchema(TypeDescription typeDescr) {
    TypeDescription result = TypeDescription.createStruct()
        .addField("operation", TypeDescription.createInt())
        .addField("originalTransaction", TypeDescription.createLong())
        .addField("bucket", TypeDescription.createInt())
        .addField("rowId", TypeDescription.createLong())
        .addField("currentTransaction", TypeDescription.createLong())
        .addField("row", typeDescr.clone());
    return result;
}

Reply:
I am thinking on simplifying this a bit by introducing extra param (isAcid) in findColumn method.

if (resultTypeDesc.getCategory() == TypeDescription.Category.STRUCT && isAcid) {
//unfold
          resultTypeDesc = findColumn(columnName, resultTypeDesc.getFieldNames(), resultTypeDesc.getChildren());
}